core/state/journal_arbitrum.go

```diff
@@ -41,8 +41,11 @@ type EvictWasm struct {
 }
 
 func (ch EvictWasm) revert(s *StateDB) {
-	asm := s.GetActivatedAsm(ch.ModuleHash) // only happens in native mode
-	CacheWasmRust(asm, ch.ModuleHash, ch.Version, ch.Debug)
+	asm, err := s.TryGetActivatedAsm(ch.ModuleHash) // only happens in native mode
+	if err == nil && len(asm) == 0 {
```

Comment:
I don't think this check is right. Maybe it should be `if err != nil || len(asm) == 0 {`?